script/prerender/contentPages.tsx

```diff
@@ -113,11 +113,11 @@ type MakeRenderPage = (services: AppOptions['services']) =>
 
 const makeRenderPage: MakeRenderPage = (services) => async({code, route}) => {
 
-  const matchState = assertDefined(
-    route.state,
-    'match state wasn\'t defined, it should have been'
-  );
-  const {bookUid, bookVersion, pageUid} = matchState;
+  if (!route.state || !('bookUid' in route.state)) {
+    throw new Error('match state wasn\'t defined, it should have been');
+  }
+
+  const {bookUid, bookVersion, pageUid} = route.state;
```

Comment:
Why did you make it this way? is it okay to hardcode the check for `bookUid` in `route.state`?

Reply:
the state type is now `{} | theStateStuff` instead of `undefined` so you can't just do a truthy or undefined check anymore. in this case in prerendering i'm requiring that the uid version of the props is supplied.

its not pretty but it works, we can try to come up with a cleaner way in the future